docs(cdp): Update Klaviyo list_profiles to reflect incremental sync support (#18338)
PostHog/posthog.com: contents/docs/cdp/sources/klaviyo.md

The `events` table is append-only, since Klaviyo events are immutable. On the initial sync, only the last 365 days of events are imported.

> The opt-in `list_profiles` table maps which profiles belong to which list as `{list_id, profile_id, joined_group_at}` rows. This is disabled by default – toggle it on in the schema configuration when setting up or editing your Klaviyo source. It supports incremental sync on `joined_group_at` (the datetime when the profile most recently joined the list). Klaviyo has no removal timestamp, so profiles removed from a list only disappear after a full refresh – incremental syncs pick up new joins and re-joins. Once synced, you can join it with your profiles table:

Reviewer: ```suggestion
The opt-in `list_profiles` table maps which profiles belong to which list as `{list_id, profile_id, joined_group_at}` rows. This is disabled by default, but can be toggled on in the schema configuration when setting up or editing your Klaviyo source. It supports incremental sync on `joined_group_at` (the datetime when the profile most recently joined the list). Incremental syncs only pick up new joins and re-joins, and will not account for profiles removed from a list. A full refresh is required if profiles need to be removed. Once synced, you can join it with your profiles table:
```
